File: grpc/src/main/java/grpc/lib/symbol/Location.java
package grpc.lib.symbol;

import org.antlr.v4.runtime.tree.TerminalNode;

public class Location {
    private int lineno;
    private int columnno;

    public Location(int lineno, int columnno) {
        this.lineno = lineno;
        this.columnno = columnno;
    }

    public Location(TerminalNode tn) {
        this(tn.getSymbol().getLine(), tn.getSymbol().getCharPositionInLine());
    }

    @Override
    public String toString() {
        return String.format("%d:%d", lineno, columnno);
    }
}

File: grpc/src/main/scala/grpc/lib/symbol/Location.scala
package grpc
package lib
package symbol

import org.antlr.v4.runtime.Token
import org.antlr.v4.runtime.tree.TerminalNode

class Location(lineno: Int) {
  def this(token: Token) = this(token.getLine)

  def this(tn: TerminalNode) = this(tn.getSymbol)

  override def toString = s"$lineno".toString
}
